apis/python/src/tiledbsoma/utils.py

```diff
@@ -0,0 +1,22 @@
+import numpy as np
+import pandas as pd
+from tiledbsoma import pytiledbsoma as clib
+from typing import Any
+import tiledb
+from .options import SOMATileDBContext
+from .options._soma_tiledb_context import _validate_soma_tiledb_context
+
+
+def build_index(keys: np.typing.NDArray[np.int64], context: SOMATileDBContext) -> clib.IntIndexer:
+    if len(np.unique(keys)) != len(keys):
+        raise pd.errors.InvalidIndexError(
+            "Reindexing only valid with uniquely valued Index objects"
+        )
+    # If we chose to keep `context` as optional, we would add this line here:
+    #     context = context or options.SOMATileDBContext()
```

Comment:
don't forget to remove these explanatory comments which were only for the review process